modules/apps/item-properties.js

```diff
@@ -13,28 +13,50 @@ export default class ItemProperties extends FormApplication {
     }
 
     constructor(...args) {
-        super(...args)
-        if (this.object.type == "weapon" || this.object.type == "ammunition"  || (this.object.type == "trait" && this.object.system.rollable.value)) {
-            this.qualities = foundry.utils.deepClone(game.wfrp4e.config.weaponQualities)
-            this.flaws = foundry.utils.deepClone(game.wfrp4e.config.weaponFlaws)
-        }
-        else if (this.object.type == "armour" || (this.object.type == "trait" && !this.object.system.rollable.value)) {
-            this.qualities = foundry.utils.deepClone(game.wfrp4e.config.armorQualities)
-            this.flaws = foundry.utils.deepClone(game.wfrp4e.config.armorFlaws)
-        }
-        else if (this.object.type == "trapping")
-        {
-            this.qualities = {}
+        super(...args);
+
+        if (ItemProperties.hasWeaponProperties(this.object)) {
+            this.qualities = foundry.utils.deepClone(game.wfrp4e.config.weaponQualities);
+            this.flaws = foundry.utils.deepClone(game.wfrp4e.config.weaponFlaws);
+        } else if (ItemProperties.hasArmourProperties(this.object)) {
+            this.qualities = foundry.utils.deepClone(game.wfrp4e.config.armorQualities);
+            this.flaws = foundry.utils.deepClone(game.wfrp4e.config.armorFlaws);
+        } else {
```

Comment:
not only `trapping`, but every other type should be handled. 

threw errors otherwise